examples/java-matter-controller/stub_src/android/annotations/Nullable.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2010 The Android Open Source Project
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.eclipse.org/org/documents/epl-v10.php
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Modified from AOSP
```

Comment:
Is it possible to include either the JSR305 or CheckerFramework annotations? The nice thing if you are to use CheckerFramework is that it supports a "non-null by default" configuration (so that everything is assumed to be non-null and enforced as such, unless it's explicitly annotated with @Nullable).

I suspect there are still some quirks around native bridging (if called from JNI I don't think it can do that enforcement) but it does have a lot more functionality than the JSR305 or Android annotations (type-use, @MonotonicNonNull, @Pure, etc.) and is natively recognized by IntelliJ.

Reply:
just use JSR305, thanks